### DISCUSSION TITLE
Feature request: Option to enable automatic video transcoding on all new imports

### DISCUSSION THREAD

**Discussion by ecker00** [photoprism/photoprism#3190] [Ideas]
Would be nice to have an option to enable automatic video transcoding on all new imports. As waiting for them on demand on a machine without hardware acceleration can take quite a while, and there is no indication showing that it's being processed.

After a while the browser times out and ends up caching the timeout out file, and says "invalid mime type" so unable to see the video after transcoding, got to clear cache/open another browser/private window to make the video play.

Just a lot of inconvenience the way it currently works.

**Comment by graciousgrey** [comment]
For now you can use the convert command to do this. Our scheduler allows you to run it automatically on a regular schedule.

https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface

https://docs.photoprism.app/user-guide/organize/video/#transcoding

**Reply by ecker00** [reply]
Was hoping to not transcode all my videos as my library is huge, mostly need it on new imports but will do this solution for now. Thank you, I'll try this.👍

**Reply by graciousgrey** [reply]
As far as I remember you can  limit the convert command to subfolders. With the latest development preview it is also possible to only transcode certain file formats.

 https://github.com/photoprism/photoprism/issues/3038

**Reply by ecker00** [reply]
Thanks for the tip, I might try to adjust the command and grab a subfolder. 👍 Seems handy to be able to configure these things from the settings UI in the future, but CLI will sort it out for me now.

**Reply by graciousgrey** [reply]
You are welcome :) We alrready have a related feature request: 
- https://github.com/photoprism/photoprism/issues/2495
